cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1157,8 +1159,12 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
         }
 
         Future<ReconciliationState> zkRollingUpdate() {
-            return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(), pod ->
-                isPodToRestart(zkDiffs.resource(), pod, zkAncillaryCmChange, this.clusterCa)
+            return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(), pod -> {
+                    if (isPodToRestart(zkDiffs.resource(), pod, zkAncillaryCmChange, this.clusterCa)) {
```

Comment:
I think `isPodToRestart()` should return a String in the same way (because that can have a lot or interesting reasons for rolling too).